code/modules/antagonists/nightmare/nightmare.dm

```diff
@@ -9,8 +9,39 @@
 	suicide_cry = "FOR THE DARKNESS!!"
 	preview_outfit = /datum/outfit/nightmare
 
+/datum/antagonist/nightmare/greet()
+	. = ..()
+	owner.announce_objectives()
+
+/datum/antagonist/nightmare/on_gain()
+	forge_objectives()
```

Comment:
why here? shouldn't this be at root?

Reply:
not sure what you mean here. i guess i just kinda copied how it was done in other places. this could probably use some refactoring

Reply:
better question:
what is forge_objectives? what does it do?
why is it on the root of /antagonist? what cases is it called in?

Reply:
it seems next to every antagonist had their own version of it, so i moved it to top level. it's just adding their objectives.

in other antags that I didn't touch, the flow is something like `on_gain() { forge objectives(); set_equipment();`

Reply:
i guess i could have root level on_gain call forge_objectives but i'm not certain about any weird outlying cases